coordinator/src/main/scala/filodb.coordinator/queryplanner/LongTimeRangePlanner.scala

```diff
@@ -53,13 +53,26 @@ import filodb.query.exec._
     val earliestRawTime = earliestRawTimestampFn
     val offsetMillis = LogicalPlanUtils.getOffsetMillis(periodicSeriesPlan)
     val (maxOffset, minOffset) = (offsetMillis.max, offsetMillis.min)
-    require(!periodicSeriesPlan.hasAtModifier || periodicSeriesPlan.getDataSource(earliestRawTime) != 2,
-      s"$periodicSeriesPlan @modifier and query range should be all greater or less than $earliestRawTime")
 
     val lookbackMs = LogicalPlanUtils.getLookBackMillis(periodicSeriesPlan).max
+
     val startWithOffsetMs = periodicSeriesPlan.startMs - maxOffset
     // For scalar binary operation queries like sum(rate(foo{job = "app"}[5m] offset 8d)) * 0.5
     val endWithOffsetMs = periodicSeriesPlan.endMs - minOffset
+    val atModifierTimestampsWithOffset = periodicSeriesPlan.atModifierTimestampsWithOffset
+
+    val isAtModifierValid = if (startWithOffsetMs - lookbackMs >= earliestRawTime) {
+      // should be in raw cluster.
+      atModifierTimestampsWithOffset.forall(at => at - lookbackMs >= earliestRawTime)
+    } else if (endWithOffsetMs - lookbackMs < earliestRawTime) {
+      // should be in down sample cluster.
+      atModifierTimestampsWithOffset.forall(at => at - lookbackMs < earliestRawTime)
+    } else {
+      atModifierTimestampsWithOffset.isEmpty
+    }
+    require(isAtModifierValid, s"@modifier $atModifierTimestampsWithOffset is not supported because it queries data" +
+      s" from both downsample and raw cluster. Please adjust the start and end time if you want to use @modifier.")
```

Comment:
This is pretty good, I think. Can you say "because it queries raw and downsampled data" (let's not mention "cluster" -- that might be confusing to anyone unfamiliar with filodb)?

Reply:
> Please adjust the start and end time

Should we give all options?:
> please adjust the query range, offsets, selector durations, and @modifier timestamps

Or maybe more generally:
> please adjust query parameters